### PR TITLE
DLPX-88349 Upgrade sshj 0.37.0 with Raul's mod dlpx73623

### PR DESCRIPTION
**Pull request by lschetanrao**
JIRA : https://delphix.atlassian.net/browse/DLPX-88349

# Description
Updating sshj with custom changes.

# Solution
This PR contains following changes 

1. Clean cherry-pick of commits from https://github.com/delphix/sshj/pull/7 on top of the latest release, 0.37.0.

Published the JAR on artifactory as version 0.37.0-dlpx73623. - [LINK](https://artifactory.delphix.com/ui/repos/tree/General/appstack-internal-services/com/hierynomus/sshj/0.37.0-dlpx73623/sshj-0.37.0-dlpx73623.jar) 
Published the POM file by copying the pom from [here](https://repo1.maven.org/maven2/com/hierynomus/sshj/0.37.0/sshj-0.37.0.pom) and replacing version from 0.37.0 to 0.37.0-dlpx732623

### CODE REVIEW

**Review by VenkatanadhanG — APPROVED** [review]
LGTM, wait for Raul's approval and merge.

**Review by rasantel — APPROVED** [review]

**Pull request merged**
